Program.cs

```diff
@@ -88,6 +110,34 @@ internal static async Task<int> Main(string[] args)
 
                 var manifest = (Manifest?) serializer.Deserialize(reader) ?? throw new InvalidDataException();
 
+                bool matching = false;
+                for (int i = 0; i < compareManifests.Count; i++)
+                {
+                    var xmlNode = compareManifests[i] ?? throw new InvalidDataException();
+                    var stream = new MemoryStream();
+                    var writer = new StreamWriter(stream);
+                    await writer.WriteAsync(xmlNode.OuterXml);
+                    await writer.FlushAsync();
+
+                    stream.Position = 0;
+                    
+                    var compareManifest = (Manifest?) serializer.Deserialize(stream) ?? throw new InvalidDataException();
+                    
+                    if (compareManifest.Name != manifest.Name) continue;

```

Comment:
This would get removed by the whole HashSet bit but I'll also note it's redundant as the first check on the next if is this and after that the loop ends.